rosbag2_cpp/src/rosbag2_cpp/clocks/time_controller_clock.cpp

```diff
@@ -200,7 +200,10 @@ bool TimeControllerClock::is_sleeping()
 
 void TimeControllerClock::wakeup()
 {
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wthread-safety-analysis"
   impl_->cv.notify_all();
+#pragma clang diagnostic pop
```

Comment:
```suggestion
#if defined(__clang__)
    #pragma clang diagnostic push
    #pragma clang diagnostic ignored "-Wthread-safety-analysis"
#endif
    // Note. We shall not lock mutex before notify_all since the notified thread would immediately
    // block again, waiting for the notifying thread to release the lock
    impl_->cv.notify_all();
#if defined(__clang__)
    #pragma clang diagnostic pop
#endif
```